src/main/java/com/snowflake/kafka/connector/internal/InternalUtils.java

```diff
@@ -170,6 +186,17 @@ static Properties createProperties(Map<String, String> conf, boolean sslEnabled)
      */
     properties.put(SFSessionProperty.ALLOW_UNDERSCORES_IN_HOST.getPropertyKey(), "true");
 
+    if (ingestionMethodConfig == IngestionMethodConfig.SNOWPIPE_STREAMING) {
+      final String providedSFRoleInConfig = conf.get(Utils.SF_ROLE);
+      if (!Strings.isNullOrEmpty(providedSFRoleInConfig)) {
+        LOGGER.debug("Using provided role {} for JDBC connection.", providedSFRoleInConfig);
+        properties.put(SFSessionProperty.ROLE, providedSFRoleInConfig);
+      } else {
+        LOGGER.debug(
+            "No role is provided, default role for the user will be used for JDBC connection.");
```

Comment:
should messages be consistent?

https://github.com/snowflakedb/snowflake-ingest-java/pull/507/files#diff-f74efdef79a994538152e7cb1924dc3620e994c9fbc6e4652e480b7bf022cdc6R135

Reply:
Ack, will do that.